src/snowflake/connector/cursor.py

```diff
@@ -272,7 +209,9 @@ def from_column(cls, col: dict[str, Any]) -> ResultMetadataV2:
 
         fields = None
         if type_code == FIELD_NAME_TO_ID["VECTOR"] and col.get("fields") is not None:
-            fields = [ResultMetadataV2Field.from_column(f) for f in col["fields"]]
+            fields = [
+                ResultMetadata.from_column({"name": None, **f}) for f in col["fields"]
```

Comment:
should this be `ResultMetadataV2`?
```suggestion
                ResultMetadataV2.from_column({"name": None, **f}) for f in col["fields"]
```